File: frontend/src/api/resumes.ts
import { del, get, patch, post } from "@/api/request";

export interface PaginatedResult<T> {
  items: T[];
  total: number;
  page: number;
  page_size: number;
  pages: number;
}

export interface ResumeListItem {
  id: string;
  title: string;
  file_name: string;
  file_type: string;
  file_size: number;
  parse_status: string;
  version: number;
  is_default: boolean;
  created_at: string;
}

export interface ResumeDetail extends ResumeListItem {
  raw_text: string | null;
  structured_data: Record<string, unknown> | null;
  parse_error: string | null;
  updated_at: string;
}

export interface ResumeUploadResponse {
  resume_id: string;
  title: string;
  file_name: string;
  file_type: string;
  file_size: number;
  parse_status: string;
}

export interface UpdateResumePayload {
  title?: string;
  is_default?: boolean;
}

export function listResumes(params?: { page?: number; page_size?: number; keyword?: string; status?: string }) {
  return get<PaginatedResult<ResumeListItem>>("/resumes", { params });
}

export function uploadResume(formData: FormData) {
  return post<ResumeUploadResponse>("/resumes/upload", formData, {
    headers: { "Content-Type": "multipart/form-data" },
  });
}

export function getResumeDetail(resumeId: string) {
  return get<ResumeDetail>(`/resumes/${resumeId}`);
}

export function updateResume(resumeId: string, payload: UpdateResumePayload) {
  return patch<ResumeDetail>(`/resumes/${resumeId}`, payload);
}

export function parseResume(resumeId: string) {
  return post<{ resume_id: string; parse_status: string }>(`/resumes/${resumeId}/parse`);
}

export function deleteResume(resumeId: string) {
  return del<null>(`/resumes/${resumeId}`);
}

File: backend/app/api/resumes.py
from uuid import UUID

from fastapi import APIRouter, Depends, File, Form, Query, UploadFile
from sqlalchemy.orm import Session

from app.core.deps import db_session, get_current_user
from app.models.user import User
from app.schemas.resume_schema import UpdateResumeRequest
from app.services.resume_service import (
    delete_resume,
    get_resume_detail,
    list_resumes,
    parse_resume,
    update_resume,
    upload_resume,
)
from app.utils.response import success_response

router = APIRouter(prefix="/resumes", tags=["resumes"])


@router.post("/upload")
def upload(
    file: UploadFile = File(...),
    title: str | None = Form(default=None),
    db: Session = Depends(db_session),
    current_user: User = Depends(get_current_user),
):
    result = upload_resume(db, current_user, file, title)
    return success_response(data=result, message="上传成功")


@router.get("")
def list_user_resumes(
    page: int = Query(default=1, ge=1),
    page_size: int = Query(default=10, ge=1, le=100),
    keyword: str | None = Query(default=None),
    status: str | None = Query(default=None),
    db: Session = Depends(db_session),
    current_user: User = Depends(get_current_user),
):
    result = list_resumes(db, current_user, page, page_size, keyword, status)
    return success_response(data=result, message="success")


@router.get("/{resume_id}")
def detail(
    resume_id: UUID,
    db: Session = Depends(db_session),
    current_user: User = Depends(get_current_user),
):
    result = get_resume_detail(db, current_user, resume_id)
    return success_response(data=result, message="success")


@router.patch("/{resume_id}")
def patch_resume(
    resume_id: UUID,
    payload: UpdateResumeRequest,
    db: Session = Depends(db_session),
    current_user: User = Depends(get_current_user),
):
    result = update_resume(db, current_user, resume_id, payload)
    return success_response(data=result, message="更新成功")


@router.delete("/{resume_id}")
def remove_resume(
    resume_id: UUID,
    db: Session = Depends(db_session),
    current_user: User = Depends(get_current_user),
):
    delete_resume(db, current_user, resume_id)
    return success_response(data=None, message="删除成功")


@router.post("/{resume_id}/parse")
def reparse_resume(
    resume_id: UUID,
    db: Session = Depends(db_session),
    current_user: User = Depends(get_current_user),
):
    result = parse_resume(db, current_user, resume_id)
    return success_response(data=result, message="解析完成")
